packages/react-vapor/src/components/badge/Badge.tsx

```diff
@@ -5,7 +5,7 @@ import {Svg} from '../svg';
 export const DEFAULT_BADGE_CLASSNAME = 'badge';
 
 export interface IBadgeProps {
-    label: string;
+    label?: string;
     icon?: string;
     extraClasses?: string[];
 }

```

Comment:
We want either:
- a label and an icon
- a label
- an icon

Right now it seems to be possible to have neither?

I would suggest something along those lines:

```typescript
export interface IBadgeWithLabelProps {
  label: string;
  icon?: string;
  extraClasses?: string[];
}
export interface IBadgeWithIconProps {
  icon: string;
  label?: string;
  extraClasses?: string[];
}
export type IBadgeProps = IBadgeWithLabelProps |  IBadgeWithIconProps;
```

Reply:
You don't have any error if you have neither? 😞 Can you have a base interface like in this answer?

https://stackoverflow.com/a/37688375

Reply:
It works! 🎉  But I had to do some type inference inside the badge implementation to be able to use the `icon`and `label` props without errors.

```ts
'label' in this.props ? <span>{this.props.label}</span> : null
```